runtime/interpreter/account_capabilities.go

```diff
@@ -0,0 +1,209 @@
+
+		case sema.AuthAccountCapabilitiesTypeGetControllerFunctionName:
+			_ = getOneControllerFunction // TODO
+		case sema.AuthAccountCapabilitiesTypeGetControllersFunctionName:
+			_ = getAllControllersFunction // TODO
+		case sema.AuthAccountCapabilitiesTypeForEachControllerFunctionName:
+			_ = forEachControllerFunction // TODO
+		case sema.AuthAccountCapabilitiesTypeIssueFunctionName:
+			_ = issueFunction // TODO
+		}
+
+		return nil
+	}
+
+	stringer := capabilitiesStringer(address, common.AuthAccountCapabilitiesStringMemoryUsage, "AuthAccount")
+
+	return NewSimpleCompositeValue(
+		gauge,
+		authAccountCapabilitiesTypeID,
+		authAccountCapabilitiesStaticType,
+		authAccountCapabilitiesFieldNames,
+		fields,
+		computeField,
+		nil,
+		stringer,
+	)
+}
+
+var publicAccountCapabilitiesFieldNames []string = nil
+var publicAccountCapabilitiesStaticType StaticType = PrimitiveStaticTypeAuthAccountCapabilities
+var publicAccountCapabilitiesTypeID = sema.PublicAccountCapabilitiesType.ID()
+
+func NewPublicAccountCapabilitiesValue(
+	gauge common.MemoryGauge,
+	address AddressValue,
+) Value {
+	getCapabilityFn := wrapGetCapabilityAsOptional(gauge, address)
+
+	fields := map[string]Value{
+		sema.AccountCapabilitiesTypeGetFunctionName:    getCapabilityFn,
+		sema.AccountCapabilitiesTypeBorrowFunctionName: newAccountCapabilitiesBorrowFunction(gauge, getCapabilityFn),
+	}
+
+	var forEachFunction *HostFunctionValue // forEach
+
+	computeField := func(name string, inter *Interpreter, locationRange LocationRange) Value {
+		switch name {
+		case sema.AccountCapabilitiesTypeForEachFunctionName:
+			if forEachFunction == nil {
+				forEachFunction = inter.newStorageIterationFunction(
+					address,
+					common.PathDomainPublic,
+					sema.PublicPathType,
+				)
+			}
+
+			return forEachFunction
+		}
+
+		return nil
+	}
+
+	stringer := capabilitiesStringer(address, common.PublicAccountCapabilitiesStringMemoryUsage, "PublicAccount")
+
+	return NewSimpleCompositeValue(
+		gauge,
+		publicAccountCapabilitiesTypeID,
+		publicAccountCapabilitiesStaticType,
+		publicAccountCapabilitiesFieldNames,
+		fields,
+		computeField,
+		nil,
+		stringer,
+	)
+}
+
+// lift the nullable result of invoking accountGetCapabilityFunction into an Optional
+func wrapGetCapabilityAsOptional(
+	gauge common.MemoryGauge,
+	address AddressValue,
+) *HostFunctionValue {
+	getCapabilityFnValue := accountGetCapabilityFunction(
+		gauge,
+		address,
+		sema.PublicPathType, // there's some discussion in FLIP #798 about changing this to a CapabilityPath
+		sema.AccountCapabilitiesTypeGetFunctionType,
+		true, // check if the capability exists
+	)
+
+	innerHostFn := getCapabilityFnValue.Function
+
+	getCapabilityFnValue.Function = func(invocation Invocation) Value {
+		maybeStorageCapabilityValue := innerHostFn(invocation)
+		if maybeStorageCapabilityValue == nil {
+			return Nil
+		}
+
+		return NewSomeValueNonCopying(
+			invocation.Interpreter,
+			maybeStorageCapabilityValue,
+		)
+	}
```

Comment:
Not sure how idiomatic this is, are there other places where we wrap HostFunctions like this?